usr/lib/tpm_stdll/tpm_specific.c

```diff
@@ -2783,6 +2942,67 @@ CK_RV token_specific_tdes_cbc(STDLL_TokData_t * tokdata,
     }
 
     return rc;
+#else
+    const EVP_CIPHER *cipher = EVP_des_ede3_cbc();
+    EVP_CIPHER_CTX *ctx = NULL;
+    CK_ATTRIBUTE *attr = NULL;
+    unsigned char dkey[3 * DES_KEY_SIZE];
+    CK_KEY_TYPE keytype;
+    CK_ULONG rc;
+    int outlen;
+
+    UNUSED(tokdata);
+
+    // get the key type
+    rc = template_attribute_find(key->template, CKA_KEY_TYPE, &attr);
+    if (rc == FALSE) {
+        TRACE_ERROR("template_attribute_find(CKA_KEY_TYPE) failed.\n");
+        return CKR_FUNCTION_FAILED;
+    }
+    keytype = *(CK_KEY_TYPE *)attr->pValue;
+
+    // get the key value
+    if (template_attribute_find(key->template, CKA_VALUE, &attr) == FALSE) {
+        TRACE_ERROR("template_attribute_find(CKA_VALUE) failed.\n");
+        return CKR_FUNCTION_FAILED;
+    }
+
+    if (keytype == CKK_DES2) {
+        memcpy(dkey, attr->pValue, 2 * DES_KEY_SIZE);
+        memcpy(dkey + (2 * DES_KEY_SIZE), attr->pValue, DES_KEY_SIZE);
+    } else {
+        memcpy(dkey, attr->pValue, 3 * DES_KEY_SIZE);
+    }
+
+    if (in_data_len % 8 || in_data_len > INT_MAX) {
```

Comment:
use constant instead of 8 and move this check before the key value is copied into the key buffer

Reply:
done